sensor/extrinsic_mapping_based_calibrator/src/extrinsic_mapping_based_calibrator.cpp

```diff
@@ -0,0 +1,1393 @@
+
+  float initial_score =
+    sourceTargetDistance<PointType>(sources, targets, initial_calibration_transform);
+  float final_score =
+    sourceTargetDistance<PointType>(sources, targets, best_multi_frame_calibration_transform);
+
+  RCLCPP_WARN(
+    this->get_logger(),
+    "Initial calibration score = %.4f (avg.squared.dist) | sqrt.score = %.4f m | discretization = "
+    "%.4f m",
+    initial_score, std::sqrt(initial_score), params_.leaf_size_dense_map_);
+  RCLCPP_WARN(
+    this->get_logger(),
+    "Best calibration score = %.4f (avg.squared.dist) | sqrt.score = %.4f m | discretization = "
+    "%.4f m",
+    final_score, std::sqrt(final_score), params_.leaf_size_dense_map_);
+  RCLCPP_WARN(
+    this->get_logger(),
+    "Test calibration score = %.4f (avg.squared.dist) | sqrt.score = %.4f m | discretization = "
+    "%.4f m",
+    best_multi_frame_calibration_multi_frame_score,
+    std::sqrt(best_multi_frame_calibration_multi_frame_score), params_.leaf_size_dense_map_);
+
+  // Publish ROS data
+  PointcloudType::Ptr initial_source_aligned_map_ptr(new PointcloudType());
+  PointcloudType::Ptr calibrated_source_aligned_map_ptr(new PointcloudType());
+
+  for (std::size_t i = 0; i < accepted_calibration_frames.size(); i++) {
+    PointcloudType::Ptr initial_tmp_ptr(new PointcloudType());
+    PointcloudType::Ptr calibrated_tmp_ptr(new PointcloudType());
+
+    pcl::transformPointCloud(
+      *sources[i], *initial_tmp_ptr,
+      accepted_calibration_frames[i].target_frame_->pose_ * initial_calibration_transform);
+    pcl::transformPointCloud(
+      *sources[i], *calibrated_tmp_ptr,
+      accepted_calibration_frames[i].target_frame_->pose_ * best_multi_frame_calibration_transform);
+
+    *initial_source_aligned_map_ptr += *initial_tmp_ptr;
+    *calibrated_source_aligned_map_ptr += *calibrated_tmp_ptr;
+  }
+
+  sensor_msgs::msg::PointCloud2 initial_source_aligned_map_msg, calibrated_source_aligned_map_msg;
+  initial_source_aligned_map_ptr->width = initial_source_aligned_map_ptr->points.size();
+  initial_source_aligned_map_ptr->height = 1;
+  calibrated_source_aligned_map_ptr->width = calibrated_source_aligned_map_ptr->points.size();
+  calibrated_source_aligned_map_ptr->height = 1;
+
+  pcl::toROSMsg(*initial_source_aligned_map_ptr, initial_source_aligned_map_msg);
+  pcl::toROSMsg(*calibrated_source_aligned_map_ptr, calibrated_source_aligned_map_msg);
+
+  initial_source_aligned_map_msg.header = accepted_calibration_frames[0].source_header_;
+  initial_source_aligned_map_msg.header.frame_id = map_frame_;
+  calibrated_source_aligned_map_msg.header = accepted_calibration_frames[0].source_header_;
+  calibrated_source_aligned_map_msg.header.frame_id = map_frame_;
+
+  initial_source_aligned_map_pub_->publish(initial_source_aligned_map_msg);
+  calibrated_source_aligned_map_pub_->publish(calibrated_source_aligned_map_msg);
+
+  response->success = false;
```

Comment:
Is this true?

Reply:
Since I do not expect this service to be used for actual code (for now), I left the `success` as false